### ISSUE TITLE
Support Edge runtime alongside Node and Browser

### ISSUE THREAD

**Issue by MaxLeiter** [algolia/search-insights.js#444]
The [getRequestForNode](https://github.com/algolia/search-insights.js/blob/886fc6191bef4fafeb8097f8ac5f37164137c5bc/lib/utils/getRequesterForBrowser.ts#L13)/Browser utilities fail when used in an [edge runtime](edge-runtime.vercel.app).

The following is logged when it's attempted: 
```
[Error: Could not find a supported HTTP request client in this environment.]
```

**Comment by Jeffrey-Zutt** [comment]
We are running into the exact same issue.

**Comment by benhinchley** [comment]
fixed by #515
